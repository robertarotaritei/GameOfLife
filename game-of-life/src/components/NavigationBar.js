import React from 'react';
import UserStore from '../stores/UserStore';
import { runInAction } from 'mobx';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Container from '@material-ui/core/Container';

class NavigationBar extends React.Component {
    constructor() {
        super();

        runInAction(() => {
            UserStore.key = sessionStorage.getItem('key');
        });
    }

    async doLogout() {
        sessionStorage.removeItem('key');
        sessionStorage.removeItem('username');
        sessionStorage.removeItem('selectedPage');
        sessionStorage.removeItem('initialState');

        runInAction(() => {
            UserStore.key = '';
            UserStore.username = '';
        })
    }

    render() {
        return (
            <AppBar className='navbar' position="fixed" style={{ background: '#063A4C', borderBottom: 'solid thin #009ECE', borderTop: 'solid thin #009ECE' }}>
                <Toolbar variant="dense" style={{ maxHeight: '1.2rem' }}>
                    <p style={{ color: '#9BE8FF' }} >
                        Welcome {UserStore.username}
                    </p>
                    <Container disableGutters>
                        <div style={{ float: 'right', color: '#9BE8FF' }}>
                            <button className='btnNav' onClick={this.props.renderDashboard} style={{ backgroundColor: this.props.selectedPage === 'dashboard' ? '#009ECE' : null }}>
                                Dashboard
                            </button>
                            <button className='btnNav' onClick={this.props.renderGameHistory} style={{ backgroundColor: this.props.selectedPage === 'history' ? '#009ECE' : null }} >
                                Game History
                            </button>
                            <button className='btnNav' onClick={this.props.renderWelcome} style={{ backgroundColor: this.props.selectedPage === 'welcome' ? '#009ECE' : null }} >
                                About
                            </button>
                            <button className='btnNav' data-testid={'logOut'} onClick={this.doLogout} style={{ marginLeft: '30px' }}>
                                Log out
                            </button>
                        </div>
                    </Container>
                </Toolbar>
            </AppBar>
        );
    }
}

export default NavigationBar;