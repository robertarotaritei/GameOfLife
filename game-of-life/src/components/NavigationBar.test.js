import React from 'react';
import { render } from '@testing-library/react';
import NavigationBar from './NavigationBar';
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";

test("NavigationBar renders without crashing", () => {
  shallow(<NavigationBar />);
});

it('NavigationBar renders correctly', () => {
  const tree = renderer
    .create(<NavigationBar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render NavigationBar Log out Button', () => {
  const { getByText } = render(<NavigationBar />);
  const linkElement = getByText(/Log out/i);
  expect(linkElement).toBeInTheDocument();
});
