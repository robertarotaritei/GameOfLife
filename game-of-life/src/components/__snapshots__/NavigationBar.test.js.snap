// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders correctly 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary navbar mui-fixed MuiPaper-elevation4"
  style={
    Object {
      "background": "#063A4C",
      "borderBottom": "solid thin #009ECE",
      "borderTop": "solid thin #009ECE",
    }
  }
>
  <div
    className="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
    style={
      Object {
        "maxHeight": "1.2rem",
      }
    }
  >
    <p
      style={
        Object {
          "color": "#9BE8FF",
        }
      }
    >
      Welcome 
      
    </p>
    <div
      className="MuiContainer-root MuiContainer-disableGutters MuiContainer-maxWidthLg"
    >
      <div
        style={
          Object {
            "color": "#9BE8FF",
            "float": "right",
          }
        }
      >
        <button
          className="btnNav"
          style={
            Object {
              "backgroundColor": null,
            }
          }
        >
          Dashboard
        </button>
        <button
          className="btnNav"
          style={
            Object {
              "backgroundColor": null,
            }
          }
        >
          Game History
        </button>
        <button
          className="btnNav"
          style={
            Object {
              "backgroundColor": null,
            }
          }
        >
          About
        </button>
        <button
          className="btnNav"
          data-testid="logOut"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "30px",
            }
          }
        >
          Log out
        </button>
      </div>
    </div>
  </div>
</header>
`;
